flamedisx/inference.py

```diff
@@ -493,6 +541,14 @@ def __init__(self, *,
         self.guess = {**bestfit,
                       **{self.target_parameter: tp_guess},
                       **self.guess}
+        self._guess_vect = self._dict_to_array(self.guess)
```

Comment:
This (and the lines below) seem duplicated in the base class __init__. Could we factor them out to a new function?